docs/reference/executionModes.md

```diff
@@ -0,0 +1,107 @@
+---
+id: executionModes
+title: Execution Modes
+---
+
+:::warning
+This page is under review and currently not visible in the menu.
+:::
+
+## Execution modes
+Execution modes select the data to be processed. By default, if you start SmartDataLakeBuilder, there is no filter applied. This means every Action reads all data from its input DataObjects.
+
+You can set an execution mode by defining attribute "executionMode" of an Action. Define the chosen ExecutionMode by setting type as follows:
+```
+executionMode {
+  type = PartitionDiffMode
+  attribute1 = ...
+}
+```
+
+### Fixed partition values filter
+You can apply a filter manually by specifying parameter `--partition-values` or `--multi-partition-values` on the command line. The partition values specified are passed to all start-Actions of a DAG and filtered for every input DataObject by its defined partition columns.
+On execution every Action takes the partition values of the input and filters them again for every output DataObject by its defined partition columns, which serve again as partition values for the input of the next Action.
+Note that during execution of the dag, no new partition values are added, they are only filtered. An exception is if you place a PartitionDiffMode in the middle of your pipeline, see next section.
+
+### PartitionDiffMode: Dynamic partition values filter
+Alternatively you can let SmartDataLakeBuilder find missing partitions and set partition values automatically by specifying execution mode PartitionDiffMode.
+
+By defining the **applyCondition** attribute you can give a condition to decide at runtime if the PartitionDiffMode should be applied or not.
+Default is to apply the PartitionDiffMode if the given partition values are empty (partition values from command line or passed from previous action).
+Define an applyCondition by a spark sql expression working with attributes of DefaultExecutionModeExpressionData returning a boolean.
+
+By defining the **failCondition** attribute you can give a condition to fail application of execution mode if true.
+It can be used to fail a run based on expected partitions, time and so on.
+The expression is evaluated after execution of PartitionDiffMode, amongst others there are attributes inputPartitionValues, outputPartitionValues and selectedPartitionValues to make the decision.
+Default is that the application of the PartitionDiffMode does not fail the action. If there is no data to process, the following actions are skipped.
+Define a failCondition by a spark sql expression working with attributes of PartitionDiffModeExpressionData returning a boolean.
+
+Example - fail if partitions are not processed in strictly increasing order of partition column "dt":
+```
+  failCondition = "(size(selectedPartitionValues) &gt 0 and array_min(transform(selectedPartitionValues, x -&gt x.dt)) &lt array_max(transform(outputPartitionValues, x -&gt x.dt)))"
```

Comment:
Is that something that you should add as subattribute of the attribute executionMode ?
Maybe add some Link to the ScalaDoc or move parts of this page to ScalaDoc?

Reply:
Don't fully understand this comment, sorry. `failCondition` is an attribute of of `PartitionDiffMode`. I assume it doesn't make sense for all execution modes.
Maybe @zzeekk can elaborate ?

Reply:
I was just not sure how I would use it in practice. Where do I need to put failcondition in my conf file ? An example would be great.

Reply:
OK, got you. Totally agree, I would also like more examples on this. I think this is a great example of what I would want to see as a blog post. Let's merge then and discuss this further separately.